File: ONLINE_CODING_TEST/school.programmers/lvl_1/42862_체육복.js
function solution(n, lost, reserve) {
  let answer = n;

  // 각 학생의 유니폼 수
  let hasUniform = new Array(n).fill(1);

  // 잃어버린 학생은 --
  for (let i = 0; i < lost.length; i++) {
    hasUniform[lost[i] - 1]--;
  }

  // 여벌이 있으면 ++
  for (let i = 0; i < reserve.length; i++) {
    hasUniform[reserve[i] - 1]++;
  }

  for (let i = 0; i < hasUniform.length; i++) {
    if (hasUniform[i] === 0) {
      if (hasUniform[i - 1] === 2) {
        hasUniform[i]++;
        hasUniform[i - 1]--;
      } else if (hasUniform[i + 1] === 2) {
        hasUniform[i]++;
        hasUniform[i + 1]--;
      } else {
        answer--;
      }
    }
  }

  return answer;
}

const tests = [{
    n: 5,
    lost: [2, 4],
    reserve: [1, 3, 5]
  },
  {
    n: 5,
    lost: [2, 4],
    reserve: [3]
  },
  {
    n: 3,
    lost: [3],
    reserve: [1]
  },
  {
    n: 7,
    lost: [2, 3, 4],
    reserve: [1, 2, 3, 6]
  },
];

for (const test of tests) {
  console.log('-'.repeat(100));
  console.log('param : ', test, '\n');
  console.time();
  console.log('\n\t->\t', solution(test.n, test.lost, test.reserve));
  console.timeEnd();
  console.timeLog();
  console.log('-'.repeat(100) + '\n');
}

File: ONLINE_CODING_TEST/school.programmers/lvl_1/42862_체육복.java
import java.util.Arrays;

// class Solution {
//   public static int solution(int n, int[] lost, int[] reserve) {
//     int answer = n - lost.length;
    
//     Arrays.sort(lost);
//     Arrays.sort(reserve);
    
//     // 여벌을 도난
//     for (int i = 0; i < lost.length; i++) {
//       for (int j = 0; j < reserve.length; j++) {
//         if (lost[i] == reserve[j]) {
//           answer++;
//           lost[i] = -1;
//           reserve[j] = -1;
//           break;
//         }
//       }
//     }
    
//     // 빌려줌
//     for (int i = 0; i < lost.length; i++) {
//       for (int j = 0; j < reserve.length; j++) {
//         // 앞이나 뒤에만 빌려줄 수 있음
//         if ((lost[i] - 1 == reserve[j]) || (lost[i] + 1 == reserve[j])) {
//           answer++;
//           reserve[j] = -1;
//           break;
//         }
//       }
//     }
    
//     return answer;
//   }
// }


//  public static int solution(int n, int[] lost, int[] reserve) {
//    int[] people = new int[n];
//    int answer = n;
//
//    for (int l : lost)
//      people[l - 1]--;
//    for (int r : reserve)
//      people[r - 1]++;
//
//    for (int i = 0; i < people.length; i++) {
//      if (people[i] == -1) {
//        if (i - 1 >= 0 && people[i - 1] == 1) {
//          people[i]++;
//          people[i - 1]--;
//        } else if (i + 1 < people.length && people[i + 1] == 1) {
//          people[i]++;
//          people[i + 1]--;
//        } else
//          answer--;
//      }
//    }
//    return answer;
//  }